pyne/alara.py

```diff
@@ -890,5 +892,242 @@ def _find_phsrc_dc(idc, phtn_src_dc):
+        s += "     4.5e5 5.1e5 5.12e5 6e5 7e5 8e5 1e6 1.33e6 1.34e6 1.5e6 1.66e6 2e6\n"
+        s += "     2.5e6 3e6 3.5e6 4e6 4.5e6 5e6 5.5e6 6e6 6.5e6 7e6 7.5e6 8e6 1e7\n"
+        s += "     1.2e7 1.4e7 2e7 3e7 5e7\nend\n"
+    s += "pulsehistory my_schedule\n"
+    s += "    1 0.0 s\nend\n"
+    s += "schedule total\n"
+    for i, irr_time in enumerate(irr_times):
+        s += "    {0} s flux_{1} my_schedule 0 s\n".format(irr_time, i)
+    s += "end\n"
+    s += "cooling\n"
+    for d in decay_times:
+        s += "    {0} s\n".format(d)
+    s += "end\n"
+    with open(input_file, 'w') as f:
+        f.write(s)
+
+def _gt_alara(data_dir, mats, neutron_spectrum, flux_magnitudes, irr_times, 
+              decay_times, num_p_groups, run_dir):
+    """
+    Function that prepares necessary input files and runs ALARA
+    
+    Parameters
+    ----------
+    data_dir : str
+        Path to nuclib file
+    mats : list of Material objects
+        List of properties of materials in the geometry
+    neutron_spectrum : list
+        Neutron spectrum (length is equal to number of n energy groups)
+    flux_magnitudes : list
+        Magnitude of flux in each neutron energy group
+    irr_times : list
+        Irradiation times [s]
+    decay_times : list
+        Decay times [s]
+    num_p_groups : int
+        Number of photon energy groups for ALARA calculation
+    run_dir : str
+        Directory to store ALARA run input and output files
 
+    Returns
+    ----------
+    phtn_src_file : str
+        Path to photon source file produced by ALARA
+    """
+    neutron_spectrum = _gt_normalize(neutron_spectrum)
+    num_n_groups = len(neutron_spectrum)
+    num_mats = len(mats)
+    num_decay_times = len(decay_times)
+    num_irr_times = len(irr_times)
+
+    # Write matlib file
+    matlib_file = os.path.join(run_dir, "matlib")
+    _gt_write_matlib(mats, matlib_file)
+
+    # Write fluxin file
+    fluxin_file = os.path.join(run_dir, "fluxin")
+    fluxes = []
+    for m in range(num_mats):
+        for n in range(num_n_groups):
+            fluxes.append([neutron_spectrum[n] if x ==
+                           n else 0 for x in range(num_n_groups)])
+        fluxes.append(neutron_spectrum) # total spectrum
+        fluxes.append([0]*175) # blank spectrum
+    _gt_write_fluxin(fluxes, fluxin_file)
+
+    # Write geom file
+    input_file = os.path.join(run_dir, "inp")
+    phtn_src_file = os.path.join(run_dir, "phtn_src")
+    # Two extra zones are needed; one for the whole spectrum
+    # and one for the zero spectrum. Total number of zones = num_n_group + 2
+    num_n_groups += 2
+    _gt_write_inp(run_dir, data_dir, mats, num_n_groups, flux_magnitudes, 
+                  irr_times, decay_times, input_file, matlib_file,
+                  fluxin_file, phtn_src_file, num_p_groups)
+
+    # Run ALARA
+    sub = subprocess.Popen(['alara',input_file], stderr=subprocess.STDOUT,
```

Comment:
PEP8: space after comma.

Reply:
Also, if you want to run a single blocking subprocess, I would recommend using `subprocess.check_call()` or `check_output()`, rather than a Popen() and communicate.